lang/cs/Org.Apache.REEF.Common/metrics/Api/IMetricTagCommonFunctions.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+namespace Org.Apache.REEF.Common.Metrics.Api
+{
+    /// <summary>
+    /// Contains functions common for Tags and <see cref="IImmutableMetric"/>. 
+    /// </summary>
+    public interface IMetricTagCommonFunctions : IMetricsInfo
+    {
+        /// <summary>
+        /// Meta-data of the metric.
+        /// </summary>
+        IMetricsInfo Info { get; }
```

Comment:
So I still wanted users to query tag.Name and tag.Desc and hence wanted these functions which are part of IMetricsInfo. On the other hand, I want to create and store IMetricsInfo object once only since I do not want to recreate it if it is asked again and again externally by application or our metrics system.

Alternatively, I can simply remove dependency on IMetricsInfo interface and have it return Name and Description as part of this interface itself.